src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -547,10 +551,324 @@ public void verifySavedFieldCannotBeRenamedReservedName() throws Exception
+                .withColumnSet(List.of(
+                        TestDataGenerator.simpleFieldDef("name", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("extraField", FieldDefinition.ColumnType.String),
+                        TestDataGenerator.simpleFieldDef("testCol", FieldDefinition.ColumnType.String)));
+        DomainResponse createResponse = dgen.createDomain(createDefaultConnection(true), "SampleSet");
+        List<Map<String, Object>> createdFields = createResponse.getColumns();
+
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "exp.materials", sampleSet);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldProperties(sampleSet);
+        DomainFieldRow variableField = domainFormPanel.addField("variableField");
+        variableField.setRecommendedVariable(true);
+        // explicitly set recommended variable false on extraField
+        DomainFieldRow extraFieldRow = domainFormPanel.getField("extraField");
+        extraFieldRow.setRecommendedVariable(false);
+
+        domainDesignerPage.clickSaveAndFinish();
+
+        DomainResponse domainResponse = dgen.getDomain(createDefaultConnection(true));
+        Map<String, Object> testColumnProperties = getPropertyPerColumn(domainResponse.getColumns(), "recommendedVariable",
+                "variableField", "extraField");
+        assertEquals("variableField should have recommendedVariable marked true", true, testColumnProperties.get("variableField"));
+        assertEquals("extraField should not have recommendedVariable marked true", false, testColumnProperties.get("extraField"));
+
+        Map<String, Object> variableCol = domainResponse.getColumns().stream()
+                .filter(a->a.get("name").equals("variableField")).findFirst().orElse(null);
+        Map<String, Object> extraFieldCol = domainResponse.getColumns().stream()
+                .filter(a->a.get("name").equals("extraField")).findFirst().orElse(null);
+        assertEquals("expect column to have recommendedVariable enabled", true, variableCol.get("recommendedVariable"));
+        assertEquals("expect column not to have recommendedVariable enabled", false, extraFieldCol.get("recommendedVariable"));
+    }
+
+
+    public Map<String, Object> getPropertyPerColumn(List<Map<String, Object>> columns, String property, String... expectedColumnNames)
+    {
+        Map<String, Object> propertyPerColumn = new HashMap<>();
+        columns.forEach(col -> propertyPerColumn.put((String) col.get("name"), col.get(property)));
+        assertThat("Didn't find expected columns", propertyPerColumn.keySet(), hasItems(expectedColumnNames));
+        return propertyPerColumn.keySet().stream()
+                .filter(key -> Arrays.asList(expectedColumnNames).contains(key))
+                .collect(Collectors.toMap(Function.identity(), propertyPerColumn::get));
+    }
```

Comment:
I know this method was my suggestion but now that I see it in use, having a single `String` argument followed by a `String` vararg makes for a confusing method call (e.g. `getPropertyPerColumn(domainResponse.getColumns(), "PHI", "notPHI", "limitedPHI", "fullPHI", "restrictedPHI")`). The `expectedColumns` would probably be better as a list.
```
    public Map<String, Object> getPropertyPerColumn(List<Map<String, Object>> columns, String property, Collection<String> expectedColumnNames)
    {
        Map<String, Object> propertyPerColumn = new HashMap<>();
        columns.forEach(col -> propertyPerColumn.put((String) col.get("name"), col.get(property)));
        assertThat("Didn't find expected columns", propertyPerColumn.keySet(), hasItems(expectedColumnNames.toArray()));
        return propertyPerColumn.keySet().stream()
                .filter(key -> expectedColumnNames.contains(key))
                .collect(Collectors.toMap(Function.identity(), propertyPerColumn::get));
    }
```